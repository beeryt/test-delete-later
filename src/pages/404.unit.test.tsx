import React from 'react'
import { render } from '@testing-library/react'

import * as gatsby from 'gatsby'
const useStaticQuery = jest.spyOn(gatsby, 'useStaticQuery')

jest.mock(
  '../components/layout',
  () =>
    function Layout({ children }: { children: React.ReactNode }) {
      return <>{children}</>
    }
)

import NotFoundPage from './404'

describe('NotFoundPage', () => {
  const siteMetadata = {
    title: 'Site Title',
    description: 'Site Description',
    siteUrl: 'example.com'
  }

  beforeAll(() => {
    useStaticQuery.mockImplementation(() => {
      return { site: { siteMetadata: siteMetadata } }
    })
  })

  afterAll(() => useStaticQuery.mockReset())

  it('renders correctly', async () => {
    render(<NotFoundPage />)
    expect(document.body).toMatchSnapshot()
  })

  it('renders not found text', async () => {
    render(<NotFoundPage />)
    expect(document.body).toHaveTextContent(/(page could )?not( be)? found/i)
  })
})
