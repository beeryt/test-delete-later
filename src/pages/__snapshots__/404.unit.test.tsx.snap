// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage renders correctly 1`] = `
<body>
  <div>
    <section
      class="section"
    >
      <div
        class="content container"
      >
        <div
          class="message is-large is-primary"
        >
          <div
            class="message-header"
          >
            <span>
              The requested page could not be found!
            </span>
          </div>
          <div
            class="message-body"
          >
            <span>
              Perhaps it has moved or been deleted.
            </span>
          </div>
        </div>
      </div>
    </section>
  </div>
</body>
`;
